lib/BlockedBy.js

```diff
@@ -114,7 +167,14 @@ class BlockedBy extends NodeResque.Plugin {
 
             // If any blocking job is running, then re-enqueue
             if (blockingBuildIds.length > 0) {
-                await this.reEnqueue(waitingKey, buildId, blockingBuildIds);
+                if (enforceBlockedBySelf && enableCollapse && collapse) {
```

Comment:
if `enforceBlockedBySelf` and `enableCollapse`/`collapse` need to be true for collapse to happen?
so if `enforceBlockedBySelf` is false and `enableCollapse`/`collapse` are true, we will still never see collapse behavior?

Reply:
I guess if blockedBy is not enabled, there will be no `waitingKeys`...

Reply:
yeah, if it's not enforceBlockedBySelf, there won't be waiting builds to collapsed

Reply:
okay 👍 

we should call this out in the guide documentation